nck/readers/mytarget_reader.py

```diff
@@ -100,6 +99,12 @@ def __check_end_posterior_to_start(self, start_date: datetime, end_date: datetim
         else:
             return True
 
+    def __get_valid_start_date_end_date(self, date_range: str, start_date: datetime, end_date: datetime):
```

Comment:
(Optional) Could we use what @ali-artefact introduced in its PR #95 to validate the input?

Reply:
We're actually already using it. I placed it as the first line of our init as if the params provided are not correct we do not go further. Is it ok for you ?

Here it is: 

```
def __init__(self, client_id, client_secret, refresh_token, request_type, date_range, start_date, end_date, **kwargs):
        check_date_range_definition_conformity(start_date, end_date, date_range)
```

Reply:
Speaking about this, what do you guys think about addin these function to utils about dates with the tests corresponding to it soon ? 

Here is what I am talking about:

```
def __check_date_input_validity(self) -> bool:
        """The goal of this function is to check the validity of the date input parameters before retrieving the data."""
        return self.__check_end_posterior_to_start(self.start_date, self.end_date) and self.__check_date_not_in_future(
            self.end_date
        )

def __check_date_not_in_future(self, end_date: datetime) -> bool:
        if end_date <= date.today():
            return True
        else:
            raise ValueError(f"The end date {end_date} is posterior to current date {date.today()}")

def __check_end_posterior_to_start(self, start_date: datetime, end_date: datetime) -> bool:
        if start_date > end_date:
            raise ValueError(f"The start date {start_date} is posterior to end date {end_date}")
        else:
            return True
```

The idea is to check if the date make sense here and not just consider their types etc...

Reply:
Yes my bad, read a bit too fast. It's ok but maybe it would be better to group all the logic around dates together in the `__init__`. So, a quick win would bin to check first and then put your line `self.start_date ...`. I think it would help readability. But again, it's optional 🙂

Reply:
Sure! Go ahead 🚀  You can follow the same design with functions that raise custom exceptions.

Reply:
Totally get you, I thought about doing this at first (like I did before). But after thinking about it I think it's cleaner this way. Let me shows you what would have to be done so that it works the same way.

```suggestion
class MyTargetReader(Reader):
    def __init__(self, client_id, client_secret, refresh_token, request_type, date_range, start_date, end_date, **kwargs):
        check_date_range_definition_conformity(start_date, end_date, date_range)
        start_date, end_date = self.__get_valid_start_date_end_date(date_range, start_date, end_date)
        self.client_id = client_id
        self.client_secret = client_secret
        self.agency_client_token = {"refresh_token": refresh_token}
        self.request_type = request_type
        self.date_range = date_range
        self.start_date = start_date
        self.start_date = end_date
        self.date_format = kwargs.get("date_format")
        self.date_are_valid = self.__check_date_input_validity()
        self.__retrieve_and_set_token()
```

Reply:
Suggestion was not the good move here but basically thought it was better to directly set start and end date from the result of the function to get start/end dates. Le me know what's better in your opinion and I'll do the change.

Reply:
Yes that's great!

Reply:
Alright, that's done. Regarding the function about the logical sense of dates provided I will create and test reusable functions later in another PR.